packages/components/src/internal/components/entities/EntityInsertGridRequiredFieldAlert.tsx

```diff
@@ -6,21 +6,24 @@ import { QueryColumn } from '../../../public/QueryColumn';
 import { Alert } from '../base/Alert';
 
 interface Props {
-    type: string;
     queryInfo: QueryInfo;
+    type: string;
 }
 
 export const EntityInsertGridRequiredFieldAlert: FC<Props> = memo(props => {
     const { type, queryInfo } = props;
-    if (!queryInfo || queryInfo.isLoading) {
-        return null;
-    }
 
-    const allRequiredCols = useMemo(() => getFieldKeysOfRequiredCols(queryInfo.getAllColumns()), [queryInfo]);
-    const insertRequiredCols = useMemo(() => getFieldKeysOfRequiredCols(queryInfo.getInsertColumns()), [queryInfo]);
-    if (allRequiredCols.length === insertRequiredCols.length) {
-        return null;
-    }
+    const allRequiredCols = useMemo(() => {
+        if (!queryInfo || queryInfo.isLoading) return [];
+
+        return getFieldKeysOfRequiredCols(queryInfo.getAllColumns());
+    }, [queryInfo]);
+
+    const insertRequiredCols = useMemo(() => {
+        if (!queryInfo || queryInfo.isLoading) return [];
+
+        return getFieldKeysOfRequiredCols(queryInfo.getInsertColumns());
+    }, [queryInfo]);
 
     const missingReqColLabels = useMemo(() => {
```

Comment:
While you're in here it looks like the dependency array for `missingReqColLabels` should be `[allRequiredCols, insertRequiredCols, queryInfo]`. Additionally, this looks like it needs to be defensive against `queryInfo` state as well.

Reply:
Updated dep array.  QueryInfo defensive check shouldn't be needed as the allRequiredcols array will be empty in that case so no operations will be done on queryInfo